crates/meroctl/src/cli/config.rs

```diff
@@ -0,0 +1,216 @@
+use std::net::IpAddr;
+
+use calimero_network::config::{BootstrapConfig, BootstrapNodes, DiscoveryConfig, SwarmConfig};
+use clap::parser::ValueSource;
+use clap::{ArgMatches, CommandFactory, Parser, ValueEnum};
+use multiaddr::Multiaddr;
+use tracing::info;
+
+use crate::cli;
+use crate::config_file::{ApplicationConfig, ConfigFile, NetworkConfig, ServerConfig, StoreConfig};
+
+/// Initialize node configuration
+#[derive(Debug, Parser)]
+pub struct ConfigCommand {
+    /// List of bootstrap nodes
+    #[arg(long, value_name = "ADDR")]
+    pub boot_nodes: Vec<Multiaddr>,
+
+    /// Use nodes from a known network
+    #[arg(long, value_name = "NETWORK", default_value = "calimero-dev")]
+    pub boot_network: Option<BootstrapNetwork>,
+
+    /// Host to listen on
+    #[arg(
+        long,
+        value_name = "HOST",
+        default_value = "0.0.0.0,::",
+        use_value_delimiter = true
+    )]
+    pub swarm_host: Vec<IpAddr>,
+
+    /// Port to listen on
+    #[arg(long, value_name = "PORT", default_value_t = calimero_network::config::DEFAULT_PORT)]
+    pub swarm_port: u16,
+
+    /// Host to listen on for RPC
+    #[arg(
+        long,
+        value_name = "HOST",
+        default_value = "127.0.0.1,::1",
+        use_value_delimiter = true
+    )]
+    pub server_host: Vec<IpAddr>,
+
+    /// Port to listen on for RPC
+    #[arg(long, value_name = "PORT", default_value_t = calimero_server::config::DEFAULT_PORT)]
+    pub server_port: u16,
+
+    /// Enable mDNS discovery
+    #[arg(long, default_value_t = true, overrides_with("no_mdns"))]
+    pub mdns: bool,
+
+    #[arg(long, hide = true, overrides_with("mdns"))]
+    pub no_mdns: bool,
+
+    /// Force edit even if the argument already exists
+    #[arg(long, short)]
+    pub force: bool,
+
+    /// Print the config file
+    #[arg(long, short)]
+    pub print: bool,
+}
+
+#[derive(Clone, Debug, ValueEnum)]
+pub enum BootstrapNetwork {
+    CalimeroDev,
+    Ipfs,
+}
+
+impl ConfigCommand {
+    pub fn run(mut self, root_args: cli::RootArgs) -> eyre::Result<()> {
+        let path = root_args.home.join(&root_args.node_name);
+
+        let matches = cli::RootCommand::command().get_matches();
+
+        let setup_matches = matches
+            .subcommand_matches("config")
+            .ok_or_else(|| eyre::eyre!("Command not recognized. Expected 'config' subcommand."))?;
+
+        let boot_network_provided = check_if_provided(&setup_matches, "boot_network");
+        let swarm_host_provided = check_if_provided(&setup_matches, "swarm_host");
+        let swarm_port_provided = check_if_provided(&setup_matches, "swarm_port");
+        let server_host_provided = check_if_provided(&setup_matches, "server_host");
+        let server_port_provided = check_if_provided(&setup_matches, "server_port");
+        let mdns_provided = check_if_provided(&setup_matches, "mdns");
```

Comment:
why this pattern? if these values are options, can't you just test if they're `Some(_)`?

reads to me like you can just make the non-option args options, and remove default values

Reply:
It was a pattern left over from when I was doing some initialisation in init, and rest in config so I needed to know if it default or provided from user.
I agree it's not needed anymore, will change to `Option<_>`